### PR TITLE
mautrix-facebook: Remove max version requirements.

### PR DESCRIPTION
**Pull request by kevincox**
It appears that these are bumped to match manual testing however they do not work well in nixpkgs as every time a dependency is updated this package breaks. To resolve this issue just strip the max version requirements and assume that it works.

Ideally we would do a semver-type comparison but this is likely not an issue because a new major version would need a new attribute name anyways.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 154067` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mautrix-facebook</li>
  </ul>
</details>

File: pkgs/servers/mautrix-facebook/default.nix [hunk 1]
```diff
@@ -32,6 +32,11 @@ python3.pkgs.buildPythonPackage rec {
     yarl
   ] ++ lib.optional enableSystemd systemd;
 
+  patchPhase = ''
+    # Drop version limiting so that every dependency update doesn't break this package.
+    sed -i -e 's/,<.*//' requirements.txt
+  '';
+
   doCheck = false;
```
Comment (kevincox) [inline pkgs/servers/mautrix-facebook/default.nix:41]: This appears to be incompatible with the new version of the mautrix library.

```
% result/bin/mautrix-facebook --version
Traceback (most recent call last):
  File "/nix/store/rppr9s436950i1dlzknbmz40m2xqqnxc-python3-3.9.9/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/nix/store/rppr9s436950i1dlzknbmz40m2xqqnxc-python3-3.9.9/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/nix/store/fcr3jf9jc5s62nw7py3q20gl8nhl3h31-python3.9-mautrix-facebook-0.3.2/lib/python3.9/site-packages/mautrix_facebook/__main__.py", line 27, in <module>
    from .user import User
  File "/nix/store/fcr3jf9jc5s62nw7py3q20gl8nhl3h31-python3.9-mautrix-facebook-0.3.2/lib/python3.9/site-packages/mautrix_facebook/user.py", line 27, in <module>
    from mautrix.bridge import BaseUser, BridgeState, async_getter_lock
ImportError: cannot import name 'BridgeState' from 'mautrix.bridge' (/nix/store/rwy0q3kfy9bp4dqqrywdz0lb9yxaz62q-python3.9-mautrix-0.14.3/lib/python3.9/site-packages/mautrix/bridge/__init__.py)
```

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 154067` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mautrix-facebook</li>
  </ul>
</details>

File: pkgs/servers/mautrix-facebook/default.nix [hunk 2]
```diff
@@ -30,9 +30,13 @@ python3.pkgs.buildPythonPackage rec {
     ruamel-yaml
     unpaddedbase64
     yarl
+    zstandard
   ] ++ lib.optional enableSystemd systemd;
 
-  doCheck = false;
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/servers/mautrix-facebook/default.nix:36]:
```suggestion
  postPatch  = ''
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**